### PR TITLE
CF Firehose logs metrics

### PR DESCRIPTION
**Pull request by ryantanjunming**
# Description
AWS firehose stream for logs and/or metrics in CF

 Fixes #

# How Has This Been Tested?
Locally using aws console and using the sam cli

# Checklist:
- [ ] I have updated the relevant component changelog(s)
- [ ] This change does not affect any particular component (e.g. it's readme or docs change)

### CODE REVIEW

File: aws-integrations/firehose/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+| RoleName | The name of the EventBridge Role |  | :heavy_check_mark: |
+| PrivateKey | Your Coralogix Private Key | |  :heavy_check_mark: |
+| CoralogixRegion | The region of your Coralogix Account | _Allowed Values:_<br>- ireland<br>- stockholm<br>- india<br>- singapore<br>- us | :heavy_check_mark: |
+| CustomUrl | Custom Coralogix url (Endpoint) |  |  |
+
+
+## Deploy the Cloudformation template
+
+```sh
+aws cloudformation deploy --template-file template.yaml --stack-name <stack_name> --capabilities CAPABILITY_NAMED_IAM  --parameter-overrides ApplicationName=<application name> SubsystemName=<subsystem name> EventbridgeStream=<EventBridge delivery stream name> RoleName=<EventBridge Role> PrivateKey=<your-private-key> CoralogixRegion=<coralogix-region> CustomUrl=<Custom Coralogix url>
```
Comment (guyrenny) [inline aws-integrations/firehose/README.md:26]: Need to change the name of PrivateKey to CoralogixApiKey and add support to CustomUrl parameter.

File: aws-integrations/firehose/template.yaml [hunk 2]
```diff
@@ -0,0 +1,668 @@
...
+    Default: ""
+  SubsystemName:
+    Type: String
+    Description: The desired Subsystem Name within the Coralogix Platform. If not set, it will be the ARN.
+    Default: ""
+  CloudwatchRetentionDays:
+    Type: Number
+    Description: Days of retention in Cloudwatch Log Groups.
+    Default: 1
+  DynamicMetadata:
```
[the remark this answers is not in the document]
Comment (ryantanjunming) [reply aws-integrations/firehose/template.yaml:32]: Haven't worked on the README yet

File: aws-integrations/firehose/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+| Parameter | Description | Default Value | Required |
+|---|---|---|---|
+| ApplicationName | Your Coralogix application name |  | :heavy_check_mark: |
+| SubsystemName | Your Coralogix Subsystem name | | :heavy_check_mark: |
+| EventbridgeStream | AWS EventBridge delivery stream name |  | :heavy_check_mark: |
+| RoleName | The name of the EventBridge Role |  | :heavy_check_mark: |
+| PrivateKey | Your Coralogix Private Key | |  :heavy_check_mark: |
+| CoralogixRegion | The region of your Coralogix Account | _Allowed Values:_<br>- ireland<br>- stockholm<br>- india<br>- singapore<br>- us | :heavy_check_mark: |
+| CustomUrl | Custom Coralogix url (Endpoint) |  |  |
```
[the remark this answers is not in the document]
Comment (ryantanjunming) [reply aws-integrations/firehose/README.md:18]: Checking with ingestion team if there is a CustomUrl for firehose, cause i don't see it in docs

File: aws-integrations/firehose/README.md (hunk 1 above)
Comment (ryantanjunming) [reply aws-integrations/firehose/README.md:23]: Will do

File: aws-integrations/firehose/template.yaml [hunk 4]
```diff
@@ -0,0 +1,668 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Transform: AWS::Serverless-2016-10-31
+Description: Send Kinesis Firehose stream data to Coralogix.
+
+Parameters:
+  CoralogixRegion:
+    Type: String
+    Description: The Coralogix location region [Europe, Europe2, India, Singapore, US]
+    AllowedValues:
```
[the remark this answers is not in the document]
Comment (ryantanjunming) [reply aws-integrations/firehose/template.yaml:9]: Checking with ingestion team if there is a US2 url for firehose, not found in documentation

File: aws-integrations/firehose/template.yaml [hunk 5]
```diff
@@ -0,0 +1,668 @@
...
+          - IntegrationTypeLogs
+          - KinesisStreamARN
+      - Label:
+          default: "Metrics Streaming"
+        Parameters:
+          - EnableMetricsStream
+          - IntegrationTypeMetrics
+          - OutputFormat
+Mappings:
+  CoralogixRegionMap:
```
[the remark this answers is not in the document]
Comment (ryantanjunming) [reply aws-integrations/firehose/template.yaml:145]: Checking with ingestion team for this

File: aws-integrations/firehose/template.yaml [hunk 6]
```diff
@@ -0,0 +1,668 @@
...
+          Resource:
+          - !Sub 'arn:aws:logs:${AWS::Region}:${AWS::AccountId}:*:*'
+      Roles:
+      - Ref: LambdaProcessorRole
+  LambdaMetricsTagsProcessors:
+    Condition: IsMetricsEnabled
+    Type: 'AWS::Lambda::Function'
+    Properties:
+      Code:
+        S3Bucket: !Sub 'cx-cw-metrics-tags-lambda-processor-${AWS::Region}'
```
[the remark this answers is not in the document]
Comment (ryantanjunming) [reply aws-integrations/firehose/template.yaml:418]: @guyrenny, i think this one can ask ingestion team to deploy to more locations when customers request for it, 
I tried making the region selectable but the lambda and s3 bucket needs to be in the same region.

The lambda enrichment is optional apparently, just improves performance.  I'll speak to matej on this when he gets back to either deploy more or give an option to disable lambda in TF and CF.

But already there are many regions available.
https://github.com/coralogix/cloudwatch-metric-streams-lambda-transformation/blob/26e3eabec2091ba6959368c5cc4a26154c08bd27/Makefile#L9C11-L9C209

File: aws-integrations/firehose/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (ryantanjunming) [reply aws-integrations/firehose/README.md]: Settled, customURL added

File: aws-integrations/firehose/template.yaml (hunk 6 above)
[the remark this answers is not in the document]
Comment (ryantanjunming) [reply aws-integrations/firehose/template.yaml:418]: wrote in README.md about the regions supported.

**Review by guyrenny — APPROVED** [review]

**Pull request merged**